### DISCUSSION TITLE
How to get -geometry to work symmetrically?

### DISCUSSION THREAD

**Discussion by SentientBeingOnAQuest** [ImageMagick/ImageMagick#4628] [Help] [answered]
The geometry setting in this command doesn't seem to be operating symmetrically:

```
magick input.jpg ^
( -background none -fill #947a27 -stroke #947a27 -strokewidth 10 -font Sylfaen ^
-pointsize 70 -size 550x -gravity center caption:"MARLINGBAITS" -blur 0x2 ) ^
-gravity Center -geometry +0+250 -composite ^
( -background none -fill #fdf9dc -stroke #fdf9dc -strokewidth 0 -font Sylfaen ^
-pointsize 140 -size 1100x -gravity center caption:"MARLINGBAITS" -resize 50% ) ^
-gravity Center -geometry +0+250 -composite ^
( -background none -fill #947a27 -stroke #947a27 -strokewidth 10 -font Sylfaen ^
-pointsize 70 -size 550x -gravity center caption:"MARLINGBAITS" -blur 0x2 ) ^
-gravity Center -geometry -0-250 -composite ^
( -background none -fill #fdf9dc -stroke #fdf9dc -strokewidth 0 -font Sylfaen ^
-pointsize 140 -size 1100x -gravity center caption:"MARLINGBAITS" -resize 50% ) ^
-gravity Center -geometry -0-250 -composite ^
output2.jpg
```

If I set all -geometry settings to 0 then both texts line up perfectly:

![output2](https://user-images.githubusercontent.com/95970582/147078775-948fa6aa-5a68-46f3-8666-11445b00793b.jpg)

Yet if I set the -geometry setting to be equidistant (as shown in the command) then they are not in fact equidistant:

![output](https://user-images.githubusercontent.com/95970582/147078884-e147f2f6-16b4-4fd3-a975-43348ef9e321.jpg)

As you can see, the top text is comparatively higher in the top box than the bottom text in the bottom box (the image is perfectly square). 

How do I make the text equidistant? And is there any way to set text position based on a percentage of the total height of the image, rather than pixels. 
(IM7Win10)

Input file:

![input](https://user-images.githubusercontent.com/95970582/147079861-d37a132d-7eb5-4fa0-b144-1019d4e4f40e.jpg)

**Comment by snibgo** [comment, marked as answer]
You create four images, and composite each over the main input. "-gravity Center -geometry +0+250" will place the centre of one created input at offset (0,250) relative to the centre of the main image. This is what happens. Similarly (0,-250). It is symmetrical.

I suggest you insert a debugging operation "+write x.png" between "-resize 50%" and ")" for an image. That image will look reasonable. Now change the background from "none" to "blue".

![x](https://user-images.githubusercontent.com/33812368/147112688-5f49fef2-ddfe-4b5e-adcf-962831a29a64.png)

That is where the asymmetry lies. To allow space for descenders "y" and "g" etc, the text is placed higher than you expect.

An obvious fix is to "-trim" each of the four images.

> And is there any way to set text position based on a percentage of the total height of the image, rather than pixels.

Instead of "+0+250", you can use fx expressions, such as "+0+%[fx:h*0.2]".

**Reply by SentientBeingOnAQuest** [reply]
Works like a charm, thanks!
